editor/plugins/script_text_editor.cpp

```diff
@@ -1736,6 +1738,43 @@ void ScriptTextEditor::_color_changed(const Color &p_color) {
 	code_editor->get_text_edit()->update();
 }
 
+void ScriptTextEditor::_update_line_change_gutters() {
+
+	if (!EditorVCSInterface::get_singleton()) {
+		return;
+	}
+	String script_path = script->get_path();
+	if (script_path.begins_with("res://")) {
+		script_path = script_path.substr(6);
+	}
+
+	Array line_diff = EditorVCSInterface::get_singleton()->get_line_diff(script_path, code_editor->get_text_edit()->get_text());
+	Map<int, int, TextEdit::GutterIndicator> line_gutters;
+
+	code_editor->get_text_edit()->clear_gutter_indicators();
+	for (int i = 0; i < line_diff.size(); i++) {
+		TextEdit::GutterIndicator indicator_type;
+		Dictionary diff = line_diff[i];
+		int new_start = diff["new_start"];
+		int new_lines = diff["new_lines"];
+		int old_lines = diff["old_lines"];
```

Comment:
All of this depends on the format of the data received from EditorVCSInterface. Could we arrange all that into a neat little struct? EditorVCSInterface can then handle transferring data from the data received from the plugin and return that to Godot

Reply:
Using data structures on both sides would be better IMO.

Reply:
@Janglee123 Yes it would be the best case, but it also needs to be convertible to-and-from a variant to be sent through the Object::call() function